src/Components/Form/Form.test.js

```diff
@@ -0,0 +1,170 @@
+         choices: ["1", "2", "3", "4", "5", "more than 5"],
+         id: 4, question: "How many people are playing?"},
+      ])
+  })
+
+  it('should render expected elements', async () => {
+
+    const mockActivities = [];
+    const mockSetActivities = jest.fn();
+    const mockUpdateActivityAnswers = jest.fn();
+    const mockSetHistory = jest.fn();
+    const mockGoBack = jest.fn();
+    const mockResetState = jest.fn();
+    
+    render(
+      <MemoryRouter>
+        <Form 
+         activities={mockActivities}
+         setActivities={mockSetActivities}
+         updateActivityAnswers={mockUpdateActivityAnswers}
+         setHistory={mockSetHistory}
+         goBack={mockGoBack}
+         resetState={mockResetState}
+        />
+      </MemoryRouter>
+    )
+
+    const question = await waitFor(() => screen.getByText('Which activities excite you right now?'))
+    expect(question).toBeInTheDocument();
+    expect(screen.getByText('music')).toBeInTheDocument();
+    expect(screen.getByText('card games')).toBeInTheDocument();
+    expect(screen.getByRole('button', {name: 'next'})).toBeInTheDocument();
+    expect(screen.getByRole('button', {name: 'back'})).toBeInTheDocument();
+  })
+
+  it('should show the next set of questions based on activities chosen', async () => {
+
+    const mockActivities = [];
+    const mockSetActivities = jest.fn();
+    const mockUpdateActivityAnswers = jest.fn();
+    const mockSetHistory = jest.fn();
+    const mockGoBack = jest.fn();
+    const mockResetState = jest.fn();
+    
+    render(
+      <MemoryRouter>
+        <Form 
+         activities={mockActivities}
+         setActivities={mockSetActivities}
+         updateActivityAnswers={mockUpdateActivityAnswers}
+         setHistory={mockSetHistory}
+         goBack={mockGoBack}
+         resetState={mockResetState}
+        />
+      </MemoryRouter>
+    )
+
+    await waitFor(() => screen.getByText('Which activities excite you right now?'))
+    userEvent.click(screen.getByText('music'));
+    userEvent.click(screen.getByText('card games'));
+    // test that this text shows up twice after clicking music
+    userEvent.click(screen.getByRole('button', {name: 'next'}));
+    const nextQuestion = await waitFor(() => screen.getByText('Which music genre(s)?'))
+    expect(nextQuestion).toBeInTheDocument();
+    expect(screen.getByText('How old is too old?')).toBeInTheDocument();
+    userEvent.click(screen.getByText('Pop'));
+    userEvent.click(screen.getByText('Country'));
+    userEvent.click(screen.getByText('1990\'s'));
+    expect(mockUpdateActivityAnswers).toHaveBeenCalledTimes(5);
+    userEvent.click(screen.getByRole('button', {name: 'next'}));
+    const gameQuestion = await waitFor(() => screen.getByText('How many people are playing?'));
+    expect(gameQuestion).toBeInTheDocument();
+    expect(mockSetHistory).toHaveBeenCalledTimes(2);
+    expect(screen.getByRole('button', {name: 'submit'})).toBeInTheDocument();
```

Comment:
These tests look very thorough, definitely checks most edge cases for our form

Reply:
Love the testing, thanks !